src/ResourceManager/DevSpaces/Commands.DevSpaces/Commands/NewAzureRmDevSpacesController.cs

```diff
@@ -0,0 +1,113 @@
+    [OutputType(typeof(PSController))]
+    public class NewAzureRmDevSpacesController : DevSpacesCmdletBase
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = "Resource Group Name.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = "DevSpaces Controller Name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            HelpMessage = "Target Resource Group Name.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string TargetResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            HelpMessage = "Target Cluster Name.")]
+        [ValidateNotNullOrEmpty]
+        public string TargetClusterName { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "A hash table which represents resource tags.")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            var msg = $"{Name} in {ResourceGroupName}";
+            if (ShouldProcess(msg, Resources.CreatingADevSpacesController))
+            {
+                RunCmdLet(NewDevSpacesControllerAction);
+            }
+        }
+
+        private void NewDevSpacesControllerAction()
+        {
+            string devSpacesNotSupportedReason = String.Empty;
+
+            WriteVerbose(string.Format(Resources.FetchCluster, TargetClusterName, TargetResourceGroupName));
+            ManagedCluster cluster = ContainerClient.ManagedClusters.Get(TargetResourceGroupName, TargetClusterName);
+            if (!cluster.IsDevSpacesSupported(out devSpacesNotSupportedReason))
+            {
+                throw new Exception(devSpacesNotSupportedReason);
+            }
+
+            WriteVerbose(string.Format(Resources.FetchClusterAccessProfile, TargetClusterName, TargetResourceGroupName));
+            ManagedClusterAccessProfile accessProfile = ContainerClient.ManagedClusters.GetAccessProfiles(TargetResourceGroupName, TargetClusterName, "clusterUser");
+            if (accessProfile == null || string.IsNullOrEmpty(accessProfile.KubeConfig))
+            {
+                throw new Exception(String.Format(Resources.CanNotFetchKubeConfig, TargetClusterName));
+            }
+
+            GenericResource resource = RmClient.Resources.Get(TargetResourceGroupName, "Microsoft.ContainerService", "", "managedClusters", TargetClusterName, "2018-03-31");
```

Comment:
Does this call return information that you don't get from ManagedClusters.Get?

Reply:
Yes, httpApplicationRouting property is not returned by ManagedClusters.Get. We need to get this property. DevSpaces only work if httpApplicationRouting is enabled in Kubernetes cluster.